stl/inc/expected

```diff
@@ -0,0 +1,1022 @@
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+// [expected.un.object]
+// clang-format off
+template <copy_constructible _Err> // TRANSITION, LWG-3688
+    requires (is_object_v<_Err> && !is_array_v<_Err> && !is_volatile_v<_Err> && !is_const_v<_Err>)
+class unexpected {
+    // clang-format on
+    static_assert(!_Is_specialization_v<_Err, unexpected>, "E must not be a specialization of unexpected");
+
+    template <class _Ty, copy_constructible _Err2> // TRANSITION, LWG-3688
+    friend class expected;
+
+public:
+    // [expected.un.ctor]
+    template <class... _Args>
+        requires is_constructible_v<_Err, _Args...>
+    constexpr explicit unexpected(in_place_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, _Args...>) // strengthened
+        : _Unexpected(_STD forward<_Args>(_Vals)...) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Err, initializer_list<_Uty>&, _Args...>
+    constexpr explicit unexpected(in_place_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+               is_nothrow_constructible_v<_Err, initializer_list<_Uty>&, _Args...>) // strengthened
+        : _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...) {}
+
+    template <class _UError = _Err>
+        requires (!is_same_v<remove_cvref_t<_UError>, unexpected> && !is_same_v<remove_cvref_t<_UError>, in_place_t> //
+                 && is_constructible_v<_Err, _UError>)
+    constexpr explicit unexpected(_UError&& _Unex) noexcept(is_nothrow_constructible_v<_Err, _UError>) // strengthened
+        : _Unexpected(_STD forward<_UError>(_Unex)) {}
+    // clang-format on
+
+    // [expected.un.observe]
+#ifdef __cpp_explicit_this_parameter // TRANSITION, P2549
+    _NODISCARD constexpr decltype(auto) error(this unexpected&& _Self) noexcept {
+        return _STD forward<unexpected>(_Self)._Unexpected;
+    }
+#else // ^^^ __cpp_explicit_this_parameter ^^^ / vvv !__cpp_explicit_this_parameter vvv
+    _NODISCARD constexpr const _Err& error() const& noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr _Err& error() & noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr const _Err&& error() const&& noexcept {
+        return _STD move(_Unexpected);
+    }
+    _NODISCARD constexpr _Err&& error() && noexcept {
+        return _STD move(_Unexpected);
+    }
+#endif // !__cpp_explicit_this_parameter
+
+    // [expected.un.swap]
+    constexpr void swap(unexpected& _Other) noexcept(is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Swap_adl(_Unexpected, _Other._Unexpected);
+    }
+
+    friend constexpr void swap(unexpected& _Left, unexpected& _Right) noexcept(
+        is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Left.swap(_Right);
+    }
+
+    // [expected.un.eq]
+    template <_Weakly_equality_comparable_with<_Err> _UErr>

```

Comment:
N4910 [expected.un.eq]/1 says *Mandates* instead of *Constraints*. I believe that this overload should be unconstrained, and (unless I'm missing something) we don't need a `static_assert` to enforce the mandate at all - simply attempting to return the required value will enforce this.